src/components/TicketList.js

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+import "./component-styles/card.css";
+import FlightTicket from "./FlightTicket";
+import { useEffect, useState } from "react/cjs/react.production.min";
+
+export default class TicketList extends React.Component {
+  state = {
+    loading: true,
+    ticket: null,
+    displayed: 0,
+    sortType: 0,
+  };
+
+  changeSortType = (type) => {
+    this.setState({ sortType: type });
+    this.updateList();
+  };
+
+  updateList = async () => {
+    let arr = [];
+    let sortedArr = [];
+
+    const res = await fetch("json/tickets.json");
+    const data = await res.json();
+    sortedArr = await sort(this.state.sortType, data);
+
+    for (let i = 0; i < 5; i++) {
+      arr.push(sortedArr[i]);
+    }
+    this.setState({ ticket: arr, loading: false });
+  };
+
+  async componentDidMount() {
+    this.updateList();
+
+    // const res = await fetch("json/tickets.json");
+    // const data = await res.json();
+    // sortedArr = await sort(this.state.sortType, data);
+
+    // for (let i = this.state.displayed; i < 5; i++) {
+    //   arr.push(sortedArr[i]);
+    //   this.setState({ displayed: i });
+    // }
+    // this.setState({ ticket: arr, loading: false });
+  }
+
+  render() {
+    return (
+      <div>
+        {this.state.loading || !this.state.ticket ? (
+          <div>Загрузка...</div>
+        ) : (
+          <div>
+            <ul>
+              {this.state.ticket.map((e) => {
+                return (
+                  <li key={e.id}>
+                    <FlightTicket ticket={e} />
+                  </li>
+                );
+              })}
+            </ul>
+          </div>
+        )}
+      </div>
+    );
+  }
+}
+
+async function sort(type, data) {
+  let arr = data;
```

Comment:
Если `data` у нас не примитив, то он передается по ссылке, таким образом, arr и data указывают на одну и ту же область памяти, то есть, на один и тот же массив.

Если идея была в иммутабельной сортировке `data`, то можно создать новую копию `data` вот так:
```suggestion
  let arr = [...data];
```